R/archive.R

```diff
@@ -0,0 +1,184 @@
+#' Archive SPI-Birds raw data
+#'
+#' @param data_folder Path. Folder containing all current SPI-Birds data (N drive)
+#' @param PopID Character string. 3-4 letter code for population.
+#' Should be the same as the folder name where data are saved.
+#' @param new_data_path File path. Updated data to use as current data in the system.
+#' Where a population uses multiple files provide a list of file paths.
+#' @param update_type Character string. Have we received a new year of data ("major")
+#' or data with bug/error fixes ("minor")?
+#'
+#' @return Returns nothing.
+#' @export
+#'
+#' @examples
+archive <- function(data_folder = choose_directory(), update_type = "major", PopID, new_data_path) {
+
+  #Force data folder and new data paths in that order
+  #This is useful if both arguments use file explorer
+  force(data_folder)
+  force(new_data_path)
+
+  #Check that PopID and new files have been provided correctly...
+  if (missing(PopID)) {
+    stop("Provide the PopID that you wish to archive")
+  }
+
+  if (missing(new_data_path) | !is.vector(new_data_path)) {
+    stop("Provide a vector of new data files.")
+  }
+
+  if (!update_type %in% c("minor", "major")) {
+    stop("Update type should be either 'major' or 'minor'")
+  }
+
+  #Extract existing file information.
+  #Find the subfolder corresponding to the population...
+  all_subfolder <- list.dirs(data_folder)
+  pop_subfolder <- all_subfolder[stringr::str_detect(all_subfolder, pattern = paste0(PopID, "_"))]
+
+  #Check that PopID matches one of the subfolders in the data
+  if (length(pop_subfolder) == 0) {
+
+    stop("No subfolder exists for this population. Do you have the correct PopID?")
+
+  }
+
+  #Find all file paths inside the folder that are considered primary data or metadata.txt
+  current_data_path <- list.files(pop_subfolder, pattern = "_PrimaryData_|MetaData.txt", full.names = TRUE)
+
+  #Separate filename from paths
+  #Need this to check that names match (regardless of the file location)
+  current_data_name <- purrr::map_chr(.x = current_data_path,
+                                      .f = ~{
+
+                                        split_data <- stringr::str_split(..1, "\\\\|/")[[1]]
+                                        filename   <- split_data[length(split_data)]
+
+                                        return(filename)
+
+                                      })
+
+  #Extract meta-data info. We need this for determining new meta-data
+  current_metadata <- utils::read.delim(current_data_path[1], sep = "", header = FALSE)
+  current_name     <- current_metadata[1, 2]
+  current_version  <- as.numeric(current_metadata[3, 2])
```

Comment:
When I ran through the code (using a local copy of the PFN data), this did not work. Specifically, one of the primary data files was read in as "current_metadata", and that messed up some of the subsequent operations. 
Now, I understand that if the POP_MetaData.txt file was already present, this may work out. But this does make me wonder if there is a more robust way to select the correct element out of "current_data_path" than just by index = 1?

Reply:
EDIT: I realised that "M" always comes before "P" in the alphabet, so if everything is named correctly, there should not be any problem here.